core/primitives/src/congestion_info.rs

```diff
@@ -326,7 +419,8 @@ impl CongestionInfoV1 {
         other_shards: &[ShardId],
         congestion_seed: u64,
     ) {
-        if self.congestion_level() < 1.0 {
+        // TODO(congestion_control) Set missed chunks count correctly.
+        if self.congestion_level(0) < 1.0 {
```

Comment:
Hm, how would this even work correctly?

I mean, here we set our own congestion info for other shards to read later. So in this code line, we cannot know how many chunks will be missed between now and then, right?
So I guess we would have to override before using it. But then it would have to be specified in the protocol, rather than giving clients the choice. :thinking:

Or, based on today's discussion, maybe we should simply never activate an allowed shard based on missing chunks, right? Then 0 works here, just needs a comment to explain the thinking.

Reply:
The only possible concern is that deadlocks come back if we deactivate round-robin.

But in my mind, that seems fine. The condition for deadlocks would be that one chunk makes no more progress at all. And what should we do in this case? Stopping traffic seems better than blowing up all shards with evergrowing numbers of receipts to buffer.